### DISCUSSION TITLE
Using Quarto with Zotero: when citing websites, 'Accessed' date missing

### DISCUSSION THREAD

**Discussion by SarahJMills** [quarto-dev/quarto-cli#9437] [Q&A]
### Description

Hi

I'm writing a dissertation in Quarto 1.3.353 (as a book project) using RStudio 2023.06.0+421.  'm using Zotero 6.0.37 with BibTeX. My University requires Harvard style referencing. At the moment, when citing websites, both in-text citations and the end-text reference list are missing the 'Accessed' field in the rendered pdf (in-text appear as 'Author, no date'...end-text as 'Author (no date)'). NB the Accessed field is definitely populated in the source Zotero library.

I've tried using harvard-cite-them-right.csl and a few other csl files from the Zotero Style Repository. I realise this might be very specific to my setup, but I've tried the University librarian who couldn't help & thought this might be a reasonable first place to look on-line.

Apologies if insufficient technical information given (I'm a clinician, not a coder), very grateful for any advice offered.

Sarah

**Comment by SarahJMills** [comment]
I think I've found the issue, but not yet sure how to fix it...
The `urldate` is missing from the bib file.  Grateful for any advice as to how to specify content of bib file.

**Reply by mcanouil** [reply]
I'm afraid, that's out of Quarto scope.

**Reply by SarahJMills** [reply]
Ahhh...apologies and thanks for putting me right.

**Reply by miah0x41** [reply]
Hi @SarahJMills,

Are you using Zotero on Windows, Mac or Linux?

I also had bib related issues and then stumbled on this plugin for Zotero which fixed most of my issues:

https://retorque.re/zotero-better-bibtex/

Since installing, it's resolved most of my issues at least making sure the bib file contains the right fields. 


Just to make sure you are having similar issues, when you open the exported bib file in a text editor does it have the "Accessed" field or whatever it's called (on a mobile at the moment so can't check)? You said it's populated within Zotero?

If it's within the `bib` file then I'd make sure the CSL actually outputs the Accessed date although it seems you've done that already.

**Reply by SarahJMills** [reply]
Very kind to you to take the trouble to reply.
I'm using Zotero on a Mac (Sonoma 14.4.1)
I do have the Better BibTeX For Zotero extension installed.
The Accessed field is populated within Zotero and when I export websites directly from Zotero, the `urldate` field is included in the resultant `.bib` file.
The issue is that the `.bib` file created in my Quarto RStudio project seems to have excluded that field.

**Reply by miah0x41** [reply]
I'm happy to attempt to try and resolve the issues you're having. 

I'm not a user of RStudio so I'm not sure I fully understand the last sentence.

My understanding is that we provide an input `bib` file and the rendering process converts that to the document format ie html. So my understanding is there is no output `bib` file. 

Apologies for being slow, but is your issue that the output website does not display the references with the Accessed Date field? Is it possible to share a screenshot? I use the `ieee-with-url.csl` format - is it worth trying with that?

For context I'm using vscode and neovim. 

My workflow with Zotero is that I export the bibliography as `references.bib` into a top level folder (e.g. assets) - the same level as `index.qmd` - in my website project. 

Within the document, in the header `yaml` section I add the two keys:

```yaml
bibliography: "assets/references.bib"
csl: ieee-with-url.csl
```

I'll try and share a screenshot tomorrow.

**Reply by SarahJMills** [reply]
My setup is similar...the RProject root folder has two .bib files: `My Library.bib` (which contains all my Zotero content) and `references.bib` (which contains anything cited in the .qmd files).

The _quarto.yml file (it's a book), includes:
`bibliography: references.bib
csl: apa.csl`

**Comment by SarahJMills** [comment]
I should add that Emiliano Heyns (retorquere), author of Better BibTeX, also tried to help on the Zotero forums.  He concluded; "...I think if you use https://quarto.org/docs/visual-editor/technical.html#citations-from-zotero, Rstudio reads the Zotero database directly, and it generates its own bibtex, no involvement of BBT. I seem to recall it is possible to configure RStudio to use BBT, but I have no experience setting that up..."

**Reply by miah0x41** [reply]
That makes sense. Sorry I couldn't be of any help.

**Reply by gordonwoodhull** [reply]
I think this makes it a https://github.com/quarto-dev/quarto issue (but I don't see how to transfer this)

**Reply by mcanouil** [reply]
Indeed, it's a Visual Editor / Quarto VSCode issue and it is already open:
- https://github.com/quarto-dev/quarto/issues/17

For Rstudio IDE issues / feature requests: https://github.com/rstudio/rstudio

**Reply by SarahJMills** [reply]
Thank you all for taking the trouble to contribute, I'll take a look at the link Mickael provided.
